samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -438,6 +443,29 @@ static bool event_handler(const struct event_header *eh)
 			/* Settings need to be loaded before advertising start */
 			start();
 			started = true;
+		} else if (event->state == MODULE_STATE_ERROR) {
+			ble_adv_stop();
```

Comment:
please create a local function for this code

Reply:
done